### PR TITLE
Ra dspdc 1560 make snapshot

### PR DESCRIPTION
**Pull request by raaidbroad**
## Why
We need a way to programmatically create snapshots for the HCA project for the sake of consistency and correctness. We will also likely want to roll this functionality into a Dagster solid so that the snapshot creation can be part of a pipeline if so desired.

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1560?atlOrigin=eyJpIjoiZTc0MDc1YTQxMzI1NGUxOThkNTI4YWQ0NTlhODc2NjMiLCJwIjoiaiJ9)
## This PR
Adds a snapshot creation function to HcaUtils, which is now renamed to HcaManage. The command line tool for checking and removing duplicates or null file refs still works (with a different name), and a snapshot option has been added. This should be able to be used in a solid the same way that the checks work currently.

### CODE REVIEW

File: orchestration/dagster_orchestration/hca_manage/manage.py [hunk 1]
```diff
@@ -23,9 +24,17 @@ def __init__(self, environment: str, project: str, dataset: str, data_repo_clien
 
         self.data_repo_client = data_repo_client
 
+        data_repo_profile_ids = {
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/manage.py:27]: ⛏️ data_repo_profile_id should be parameterized so they can be passed in by the calling solid at runtime

File: orchestration/dagster_orchestration/hca_manage/manage.py [hunk 2]
```diff
@@ -191,6 +200,32 @@ def submit_soft_delete(self, target_table: str, target_path: str) -> str:
 
         return response.json()["id"]
 
+    def submit_snapshot_request(self, optional_qualifier: Optional[str] = None) -> str:
+        date_stamp = str(datetime.today().date()).replace("-", "")
+        if optional_qualifier:
+            # prepend an underscore if this string is present
+            optional_qualifier = f"_{optional_qualifier}"
+        snapshot_name = f"{self.dataset}___{date_stamp}{optional_qualifier}"
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/manage.py:208]: ⛏️ It'd be helpful to log this out as it's useful when debugging with Jade if something goes awry.

File: orchestration/dagster_orchestration/hca_manage/manage.py [hunk 3]
```diff
@@ -191,6 +200,32 @@ def submit_soft_delete(self, target_table: str, target_path: str) -> str:
...
+
+        reader_list = {
+            "dev": ["hca-snapshot-readers@dev.test.firecloud.org"],
+            "prod": ["hca-snapshot-readers@firecloud.org"]
+        }
+
+        response = self.data_repo_client.create_snapshot(
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/manage.py:215]: ⛏️ In fact, I think it'd be useful to output the entire payload we're submitting to Jade (i.e., so we can reconstruct a curl if something goes wrong).

File: orchestration/dagster_orchestration/hca_manage/main.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        self.print_help()
+        sys.exit(2)
+
+
+def get_api_client(host: str) -> RepositoryApi:
+    # create API client
+    config = Configuration(host=host)
+    config.access_token = default_google_access_token()
+    client = ApiClient(configuration=config)
+    client.client_side_validation = False
```
[the remark this answers is not in the document]
Comment (raaidbroad) [reply orchestration/dagster_orchestration/hca_manage/main.py:24]: They aren't, I tried :/

File: orchestration/dagster_orchestration/hca_manage/manage.py [hunk 5]
```diff
@@ -23,9 +25,12 @@ def __init__(self, environment: str, project: str, dataset: str, data_repo_clien
 
         self.data_repo_client = data_repo_client
 
+        self.data_repo_profile_id = data_repo_profile_id
+
         self.filename_template = f"sd-{project}-{dataset}-{{table}}.csv"
 
-        self.bigquery_client = bigquery.Client(project=self.project)
+        if self.project:
+            self.bigquery_client = bigquery.Client(project=self.project)
```
[the remark this answers is not in the document]
Comment (raaidbroad) [reply orchestration/dagster_orchestration/hca_manage/manage.py:33]: Can you explain what you mean by initializing it lazily? I'm unfamiliar with how to do that

**Review by aherbst-broad — APPROVED** [review]

**Review by ThePletch — APPROVED** [review]
:boat:

**Pull request merged**